File: 643-find-max-average/Solution.java
import java.util.Arrays;

class Solution643 {
    static double findMaxAverage(int[] nums, int k) {
        int left = 0;
        int windowSum = 0;
        for(int i = 0; i < k; i++){
            windowSum += nums[i];
        }
        int maxVal = windowSum;
        for (int index = k-1; index < nums.length; index++) {
            if (index > k-1)
                windowSum += nums[index];
            maxVal = Math.max(maxVal, windowSum);
            windowSum -= nums[left];
            left++;
        }
        return (double)maxVal/k;
    }
    public static void main(String[] args) {
        int[] nums = {1,12,-5,-6,50,3};
        int k = 4;
        System.out.println("Ans = " + findMaxAverage(nums, k));
    }
}

File: 643-find-max-average/solution.go
package main

import (
	"fmt"
	gft "gopractice/gofunctool"
)

func findMaxAverage(nums []int, k int) float64 {
	maxVal := gft.ArrSumInt(nums[:k])
	windowSum := maxVal

	for left, index := 0, k-1; index < len(nums); left, index = left+1, index+1 {
		if index > k-1 {
			windowSum += nums[index]
		}
		maxVal = max(maxVal, windowSum)
		windowSum -= nums[left]
	}
	return float64(maxVal) / float64(k)
}

func main() {
	nums := []int{1, 12, -5, -6, 50, 3}
	k := 4
	fmt.Println("Ans = ", findMaxAverage(nums, k))
}
